extractor/src/test/java/org/schabi/newpipe/extractor/services/youtube/YoutubeThrottlingDecrypterTest.java

```diff
@@ -8,49 +8,72 @@
+        for (final String encryptedUrl : encryptedUrls) {
+            final String decryptedUrl = YoutubeThrottlingDecrypter.apply(encryptedUrl,
+                    "jE1USQrs1rw");
+            // The cipher function changes over time, so we just check if the n param changed.
+            assertNotEquals(encryptedUrl, decryptedUrl);
+        }
     }
 
     @Test
-    public void testDecode__noNParam__success() throws ParsingException {
-        final String noNParamUrl = "https://r5---sn-4g5ednsz.googlevideo.com/videoplayback?expire=1626553257&ei=SefyYPmIFoKT1wLtqbjgCQ&ip=127.0.0.1&id=o-AIT5xGifsaEAdEOAb5vd06J9VNtm-KHHolnaZRGPjHZi&itag=140&source=youtube&requiressl=yes&mh=xO&mm=31%2C29&mn=sn-4g5ednsz%2Csn-4g5e6nsr&ms=au%2Crdu&mv=m&mvi=5&pl=24&initcwndbps=1322500&vprv=1&mime=audio%2Fmp4&ns=cA2SS5atEe0mH8tMwGTO4RIG&gir=yes&clen=3009275&dur=185.898&lmt=1626356984653961&mt=1626531173&fvip=5&keepalive=yes&fexp=24001373%2C24007246&beids=23886212&c=WEB&txp=6411222&sparams=expire%2Cei%2Cip%2Cid%2Citag%2Csource%2Crequiressl%2Cvprv%2Cmime%2Cns%2Cgir%2Cclen%2Cdur%2Clmt&lsparams=mh%2Cmm%2Cmn%2Cms%2Cmv%2Cmvi%2Cpl%2Cinitcwndbps&lsig=AG3C_xAwRgIhAPueRlTutSlzPafxrqBmgZz5m7-Zfbw3QweDp3j4XO9SAiEA5tF7_ZCJFKmS-D6I1jlUURjpjoiTbsYyKuarV4u6E8Y%3D&sig=AOq0QJ8wRQIgRD_4WwkPeTEKGVSQqPsznMJGqq4nVJ8o1ChGBCgi4Y0CIQCZT3tI40YLKBWJCh2Q7AlvuUIpN0ficzdSElLeQpJdrw==";
+    void testDecode__noNParam__success() throws ParsingException {
+        final String noNParamUrl = "https://rr5---sn-hgn7rn7r.googlevideo.com/videoplayback?expire=1660678263&ei=F5z7YtiqKKjkx_AP1qSFuAg&ip=127.0.0.1&id=o-AG-ATci4w870PV5NXT2n_qeXKNL7O7kIHcRb1kiQSPfZ&itag=597&source=youtube&requiressl=yes&mh=_u&mm=31%2C26&mn=sn-hgn7rn7r%2Csn-4g5edn6r&ms=au%2Conr&mv=m&mvi=5&pl=17&initcwndbps=1091250&spc=lT-KhvhRm_zaHWUnpn3dKQEpkBdYuEs&vprv=1&svpuc=1&mime=video%2Fmp4&gir=yes&clen=3837282&dur=1010.080&lmt=1606602836536371&mt=1660656311&fvip=1&keepalive=yes&fexp=24001373%2C24007246&c=ANDROID&rbqsm=fr&txp=5532434&sparams=expire%2Cei%2Cip%2Cid%2Citag%2Csource%2Crequiressl%2Cspc%2Cvprv%2Csvpuc%2Cmime%2Cgir%2Cclen%2Cdur%2Clmt&sig=AOq0QJ8wRQIgRt7PGHGTgzK_GNzjaujZY0p9_0nLpmXOD_4xChR7Ve8CIQCcIzjnAyjbiNDMd7-nx3dNC5K-qxQyHIvakWjoQfRtew%3D%3D&lsparams=mh%2Cmm%2Cmn%2Cms%2Cmv%2Cmvi%2Cpl%2Cinitcwndbps&lsig=AG3C_xAwRAIgI7Ewc_vB43w_3gNIlV3eU_AXnNwFqjMJQVzi3DDYCJkCIGXo54Urb37uUnt502tLOlJ7jZXo-enGe2a_LZDX2T8D";
         final String decrypted = YoutubeThrottlingDecrypter.apply(noNParamUrl, "jE1USQrs1rw");
 
         assertEquals(noNParamUrl, decrypted);
     }
 
+    @Test
+    void testCacheCountAndClearCache__success() throws ParsingException {
```

Comment:
This test is not working:
```
org.opentest4j.AssertionFailedError: expected: <3> but was: <21>
	at org.junit.jupiter.api.AssertionFailureBuilder.build(AssertionFailureBuilder.java:151)
	at org.junit.jupiter.api.AssertionFailureBuilder.buildAndThrow(AssertionFailureBuilder.java:132)
	at org.junit.jupiter.api.AssertEquals.failNotEqual(AssertEquals.java:197)
	at org.junit.jupiter.api.AssertEquals.assertEquals(AssertEquals.java:150)
	at org.junit.jupiter.api.AssertEquals.assertEquals(AssertEquals.java:145)
	at org.junit.jupiter.api.Assertions.assertEquals(Assertions.java:527)
	at org.schabi.newpipe.extractor.services.youtube.YoutubeThrottlingDecrypterTest.testCacheCountAndClearCache__success(YoutubeThrottlingDecrypterTest.java:68)
```

because it's not stateless...

<sub>Yeah caching in static methods is a great idea...</sub>

Reply:
Test changes have been dropped.